datafusion/src/physical_plan/expressions/min_max.rs

```diff
@@ -38,6 +38,15 @@ use arrow::{
 
 use super::format_state_name;
 
+// min/max aggregation only returns a value for each group
+// and should not be a Dictionary data type anymore
+fn min_max_aggregate_data_type(input_type: DataType) -> DataType {
+    match input_type {
+        DataType::Dictionary(_, value_type) => (*value_type).clone(),
+        _ => input_type.clone(),
+    }
```

Comment:
I think you can save the clone by reusing the types passed in. Something like (untested)


```suggestion
    if let DataType::Dictionary(_, value_type) = input_type {
        value_type
    } else {
        input_type
    }
```

I doubt the extra clone makes any noticeable difference, but I figured I would point it out